Coverity fixes

ldorau opened the pull request:
Fix issues detected by Coverity

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/pmem/vltrace/29)
<!-- Reviewable:end -->

krzycz commented:
Reviewed 6 of 7 files at r1.
Review status: 6 of 7 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/bpf_ctx.c, line 367 at r1](https://reviewable.io:443/reviews/pmem/vltrace/29#-KuyRwadEvsSLBf71UwI:-KuyRwadEvsSLBf71UwJ:bjdia6s) ([raw file](https://github.com/pmem/vltrace/blob/05501147c2e7a411fc8ad6d9a4f58e7a889ac5e9/src/bpf_ctx.c#L367)):*
> ```C
> 	}
> 
> 	enum perf_reader_type_t perf_reader_type =
> ```

What's the problem here?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/vltrace/29)*
<!-- Sent from Reviewable.io -->

krzycz commented:
Reviewed 1 of 7 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/vltrace/29)*
<!-- Sent from Reviewable.io -->

ldorau commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[src/bpf_ctx.c, line 367 at r1](https://reviewable.io:443/reviews/pmem/vltrace/29#-KuyRwadEvsSLBf71UwI:-KuySx-Yaj2latM5iU6c:b-roa0it) ([raw file](https://github.com/pmem/vltrace/blob/05501147c2e7a411fc8ad6d9a4f58e7a889ac5e9/src/bpf_ctx.c#L367)):*
<details><summary><i>Previously, krzycz (Krzysztof Czurylo) wrote…</i></summary><blockquote>

What's the problem here?
</blockquote></details>

The type of the last argument of ```append_item_to_pr_arr()``` (below) is ```enum perf_reader_type_t``` and not ```enum bpf_prog_type```

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/vltrace/29)*
<!-- Sent from Reviewable.io -->

krzycz commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[src/bpf_ctx.c, line 367 at r1](https://reviewable.io:443/reviews/pmem/vltrace/29#-KuyRwadEvsSLBf71UwI:-KuyUngbtWFD_GeJgivp:b-r8969a) ([raw file](https://github.com/pmem/vltrace/blob/05501147c2e7a411fc8ad6d9a4f58e7a889ac5e9/src/bpf_ctx.c#L367)):*
<details><summary><i>Previously, ldorau (Lukasz Dorau) wrote…</i></summary><blockquote>

The type of the last argument of ```append_item_to_pr_arr()``` (below) is ```enum perf_reader_type_t``` and not ```enum bpf_prog_type```
</blockquote></details>

So, it looks like there is some problem with the design. Do we have two enums, where one is a subset of the another one?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/vltrace/29)*
<!-- Sent from Reviewable.io -->

ldorau commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[src/bpf_ctx.c, line 367 at r1](https://reviewable.io:443/reviews/pmem/vltrace/29#-KuyRwadEvsSLBf71UwI:-Kuy_XPvwlTvDeLiE5Or:b2nhswy) ([raw file](https://github.com/pmem/vltrace/blob/05501147c2e7a411fc8ad6d9a4f58e7a889ac5e9/src/bpf_ctx.c#L367)):*
<details><summary><i>Previously, krzycz (Krzysztof Czurylo) wrote…</i></summary><blockquote>

So, it looks like there is some problem with the design. Do we have two enums, where one is a subset of the another one?
</blockquote></details>

Yes

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/vltrace/29)*
<!-- Sent from Reviewable.io -->

ldorau commented:
Review status: 6 of 8 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/bpf_ctx.c, line 367 at r1](https://reviewable.io:443/reviews/pmem/vltrace/29#-KuyRwadEvsSLBf71UwI:-Kv6MOSzlaPAHVUHKY_w:b-886gsx) ([raw file](https://github.com/pmem/vltrace/blob/05501147c2e7a411fc8ad6d9a4f58e7a889ac5e9/src/bpf_ctx.c#L367)):*
<details><summary><i>Previously, ldorau (Lukasz Dorau) wrote…</i></summary><blockquote>

Yes
</blockquote></details>

Fixed

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/vltrace/29)*
<!-- Sent from Reviewable.io -->

ldorau commented:
Review status: 6 of 8 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/bpf_ctx.c, line 367 at r1](https://reviewable.io:443/reviews/pmem/vltrace/29#-KuyRwadEvsSLBf71UwI:-Kv6OTqDS3Y09SqL8YJU:b8fcpz8) ([raw file](https://github.com/pmem/vltrace/blob/05501147c2e7a411fc8ad6d9a4f58e7a889ac5e9/src/bpf_ctx.c#L367)):*
<details><summary><i>Previously, ldorau (Lukasz Dorau) wrote…</i></summary><blockquote>

Fixed
</blockquote></details>

@krzycz I fixed it, please review the last changes

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/vltrace/29)*
<!-- Sent from Reviewable.io -->

krzycz commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 2 of 2 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/vltrace/29#-:-Kv6WlmDDbP32J_eo1lM:bnfp4nl)*
<!-- Sent from Reviewable.io -->

ldorau commented:
It is better to merge it before PR #26.

marcinslusarz commented on tools/bench_sc/bench_sc.c:
```diff
@@ -106,7 +110,7 @@ main(int argc, char *argv[])
 
 	iters_qty = atol(argv[1]);
-	if (iters_qty == 0) {
+	if (iters_qty <= 0) {
```
iters_qty is unsigned

ldorau replied:
Indeed

ldorau replied:
Fixed

marcinslusarz commented on test/test_syscalls.c:
```diff
@@ -174,52 +174,61 @@ test_basic_syscalls(void)
-	fstat(fd, &buf);
+	(void) fstat(fd, &buf);
 s();
 	close(fd);
```
Why do you exclude close?

ldorau replied:
Because it is not considered as an issue by Coverity. How often do you check the return value of close()?

ldorau replied:
Fixed

krzycz commented:
Reviewed 2 of 2 files at r3.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/vltrace/29)*
<!-- Sent from Reviewable.io -->

marcinslusarz commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/vltrace/29#-:-KvCgTx47X5eRNfMJEZ6:bnfp4nl)*
<!-- Sent from Reviewable.io -->
